tests/bats/test_tcp_server.c

```diff
@@ -0,0 +1,35 @@
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+
+int main(int argc, char *argv[])
+{
+	int listenfd = 0, connfd = 0;
+	struct sockaddr_in serv_addr; 
+
+	listenfd = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
Check retval for errors for calls (plus others below)

Reply:
done